tests/src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -385,6 +385,120 @@ TEST_F(TopoShapeExpansionTest, MapperHistoryGenerated)
+
+    // A TopoShape without a map that will be replaced by another map
+    auto shapeWithoutMapAfterReset {TopoShape(shapeWithoutMap)};
+
+    // A TopoShape with a map
+    auto shapeWithMap {
+        TopoShape(BRepBuilderAPI_MakeEdge(gp_Pnt(0.0, -1.0, 0.0), gp_Pnt(0.0, 1.0, 0.0)).Edge(),
+                  3)};
+    shapeWithMap.makeShapeWithElementMap(shapeWithMap.getShape(),
+                                         TopoShape::Mapper(),
+                                         {shapeWithMap});
+
+    // A TopoShape with a map that will be replaced by another map
+    auto shapeWithMapAfterReset {TopoShape(shapeWithMap)};
+    shapeWithMapAfterReset.makeShapeWithElementMap(shapeWithMapAfterReset.getShape(),
+                                                   TopoShape::Mapper(),
+                                                   {shapeWithMapAfterReset});
+
+    // A TopoShape with a map that will be replaced by an empty map
+    auto shapeWithMapAfterEmptyReset {TopoShape(shapeWithMap)};
+    shapeWithMapAfterEmptyReset.makeShapeWithElementMap(shapeWithMapAfterEmptyReset.getShape(),
+                                                        TopoShape::Mapper(),
+                                                        {shapeWithMapAfterEmptyReset});
+
+    // A new map
+    auto newElementMapPtr {std::make_shared<Data::ElementMap>()};
+    newElementMapPtr->setElementName(IndexedName("Edge", 2),
+                                     MappedName("Edge2;:H,E"),
+                                     3,
+                                     nullptr,
+                                     true);
+
+    // Act
+    shapeWithoutMapAfterReset.resetElementMap(newElementMapPtr);
+    shapeWithMapAfterReset.resetElementMap(newElementMapPtr);
+    shapeWithMapAfterEmptyReset.resetElementMap(nullptr);
+
+    // Assert
+    // Check that the original maps haven't been modified
+    EXPECT_EQ(shapeWithoutMap.getElementMapSize(false), 0);
+    EXPECT_EQ(shapeWithMap.getElementMapSize(false), 3);
+
+    // Check that the two shapes have the same map
+    EXPECT_EQ(shapeWithoutMapAfterReset.getElementMap(), shapeWithoutMapAfterReset.getElementMap());
+    // Check that inside the shape's map there's the element of the new map (same result if
+    // checking with the other shape)
+    EXPECT_NE(shapeWithoutMapAfterReset.getElementMap()[0].name.find("Edge2"), -1);
+    // Check that there aren't leftovers from the previous map
+    EXPECT_EQ(shapeWithMapAfterReset.getElementMap()[0].name.find("Edge1"), -1);
+
+    // Check that the map has been emptied
+    EXPECT_EQ(shapeWithMapAfterEmptyReset.getElementMapSize(false), 0);
+}
+
+TEST_F(TopoShapeExpansionTest, flushElementMapTest)
+{
+    // Arrange
+    // Creating various TopoShapes to check different conditions
+
+    // A TopoShape with a map that won't be flushed
+    auto shapeWithtMapNotFlushed {
+        TopoShape(BRepBuilderAPI_MakeEdge(gp_Pnt(-1.0, 0.0, 0.0), gp_Pnt(1.0, 0.0, 0.0)).Edge(),
```

Comment:
Witht -> With ?

Reply:
Yes of course, typo.

Fixing it in the next commit